### ISSUE TITLE
Goldendict Ctrl+c+c shortkey cause Ctrl+c stop working for other applications in Windows 8.1

### ISSUE THREAD

**Issue by exlons** [goldendict/goldendict#427]
When use Goldendict with shortkey Ctrl+c+c set for translate word in clipboard, Ctrl+c will stop working for other apllication, in my case, Eclipse/Netbeans. when  exit GoldenDict, the Ctrl+c shortkey will back to working for the other application.

Operating system, Windows 8.1
GoldenDict ver.1.5.0-RC-210-gdd84c10

**Comment by lionelster** [comment]
Having the same problem on Linux Mint. Disabling or changing the hotkey doesn't help either.

**Comment by janot** [comment]
Same problem on Win 10. 
"Run as administrator" helps.

**Comment by xiaoyifang** [comment]
this issue is still existed.

**Comment by Chaoses-Ib** [comment]
It's caused by [User Interface Privilege Isolation](https://github.com/Chaoses-Ib/Windows/blob/main/Kernel/Security/UIPI.md) of Windows. UIPI will block Send/PostMessage to windows of higher integrity level. However, it won't block hotkeys registered by `RegisterHotKey()` which I think GoldenDict is using. So GoldenDict can receive your Ctrl+C hotkey, but can't send copy messages (WM_COPY or Ctrl+C via WM_KEY/SendInput) to the original window which is of a higher integrity level.

"Run as administrator" is a way to elevate the integrity level of GoldenDict. Other solutions exist, but "Run as administrator" should be the best way.

**Comment by nonwill** [comment]
> "Run as administrator" should be the best way.

A word of caution, don't do it! 
Don't expose your system to the risk of **_buffer overflow_** vulnerabilities caused by GD!

**Comment by Chaoses-Ib** [comment]
The typical process for retrieving text from the clipboard is:
```cpp
OpenClipboard(hwnd);
HANDLE hdata = GetClipboardData(CF_TEXT);
char* text = static_cast<char*>(GlobalLock(hData));
...
GlobalUnlock(hdata);
CloseClipboard();
```
As you can see, there is no fixed-size buffer, and hence no buffer overflow is possible. Why are you worried about buffer overflows? Has GD ever had a buffer overflow vulnerability about the clipboard?

**Comment by nonwill** [comment]
> Why are you worried about buffer overflows? Has GD ever had a buffer overflow vulnerability about the clipboard?

Just because it is not available here does not mean it is not available elsewhere. So don't be frivolous and let app run as administrator.

**Comment by vedgy** [comment]
For what it is worth, running a GUI application as root (GNU/Linux equivalent of administrator) is considered a very bad idea. See e.g. https://bugzilla.gnome.org//show_bug.cgi?id=772875#c5

**Comment by Chaoses-Ib** [comment]
I agree that a GUI program shouldn't be run as elevated. I have added a security warning to my reply. But this issue is really annoying, and I will still use "Run as administrator" to fix it for myself. If you are willing to fix this issue by thread impersonating, that's surely better and I will be glad to see that.
